File: c/boj/1476.c
// 날짜 계산
// https://www.acmicpc.net/problem/1476

#include <stdio.h>

int main(void)
{
    int e, s, m;
    int i = 1;
    int j = 1;
    int k = 1;
    
    scanf("%d %d %d", &e, &s, &m);
    int count = 1;

    while(1) {
        if (e == i && j == s && k == m) break;

        i++;
        j++;
        k++;
        if (i == 16) i = 1;
        if (j == 29) j = 1;
        if (k == 20) k = 1;

        count++;
    }

    printf("%d\n", count);
    return 0;
}

File: javascript/boj/1476.js
// 날짜 계산
// https://www.acmicpc.net/problem/1476

function gcd(a, b) {
    let c;
    while (b) {
        c = a % b;
        a = b;
        b = c;
    }

    return a;
}

function lcm(a, b) {
    return (a * b) / gcd(a, b);
}

function getLCM(arr) {
    let cur = arr[0];

    for (let i = 1; i < arr.length; i++) {
        cur = lcm(cur, arr[i]);
    }

    return cur;
}

let stdin = "input.txt";
// stdin = "/dev/stdin";

let input = require("fs")
    .readFileSync(stdin)
    .toString()
    .trim()
    .split(" ")
    .map(Number);

let [E, S, M] = input;
let year = 0;
while (++year) {
    if ((year - E) % 15 === 0 && (year - S) % 28 === 0 && (year - M) % 19 === 0)
        break;
}

console.log(year);
